packages/app-cli/app/command-mkbook.test.ts

```diff
@@ -0,0 +1,32 @@
+import { setupDatabaseAndSynchronizer, switchClient } from '@joplin/lib/testing/test-utils';
+import { setupCommandForTesting, setupApplication } from './utils/testUtils';
+const Command = require('./command-mkbook');
+
+
+describe('command-mkbook', function() {
+
+	beforeEach(async () => {
+		await setupDatabaseAndSynchronizer(1);
+		await switchClient(1);
+		await setupApplication();
+	});
+
+	const command = setupCommandForTesting(Command);
+
+	it('should create a subfolder in first folder', async () => {
+		await expect(command.action({ 'new-notebook': 'folder1', options: { s: false, sub: false } })).resolves.not.toThrowError();
+		await expect(command.action({ 'new-notebook': 'folder1.1', options: { s: true, sub: true } })).resolves.not.toThrowError();
+	});
+
+	it('should create a subfolder in the second destination folder', async () => {
+		await expect(command.action({ 'new-notebook': 'folder2', options: { s: false, sub: false } })).resolves.not.toThrowError();
+		await expect(command.action({ 'new-notebook': 'folder2.1', 'notebook': 'folder2', 'options': { s: true, sub: true } })).resolves.not.toThrowError();
+	});
+
+	it('should not be possible to create subfolder in ambiguous destination folder', async () => {
+		await expect(command.action({ 'new-notebook': 'folder3', options: { s: false, sub: false } })).resolves.not.toThrowError();
+		await expect(command.action({ 'new-notebook': 'folder3', options: { s: false, sub: false } })).resolves.not.toThrowError();	// ambiguous folder
+		await expect(command.action({ 'new-notebook': 'folder3.1', 'notebook': 'folder3', 'options': { s: true, sub: true } })).rejects.toThrowError();
+	});
```

Comment:
None of these tests are testing functionalities. There should be at least one test that creates a subnotebook and verifies that it has been created. Is it not obvious actually? If it's difficult implement the test, please bring it up in the discussion instead of creating tests that do nothing.

Reply:
I admit that creating tests the way you expect is difficult for me. (Maybe I don't understand something correctly) - Also I would like to discuss how it should be done correctly for the next time. Thanks for your offer!

From my point of view the tests work correctly. 
If someone changes the function logic of "mkbooks", e.g. so that subnotebooks cannot be created, then it fails, if the check for duplicate entries is deleted then it fails or someone mistypes while correcting the variable name it fails also.

Is there any chance that you (or someone else who happens to read this message) can show me how to do it on my test? I have to confess that I didn't really figer out how to do this from the [clue](https://github.com/laurent22/joplin/blob/dev/packages/app-cli/app/command-done.test.ts)

Thanks for the effort in advance!

Reply:
The test needs to check that the command actually does something. With your test if the command does nothing, or something unrelated, the test will still pass. So you need for example to create a sub-notebook, then check that it has been created, with the correct title, and that its parent_id is correct, etc.

The clue you mention shows an example of doing this.

Reply:
I think I have understood it now. Correct me if my thought process is wrong. Now there is a double check. First, with 'await expect(command.action({ 'new-notebook': 'folder1' })).resolves.not.toThrowError();' I check the command first to see if I get an error message from there already.

Second, what you were getting at is that regardless of the command, I should also check if a folder was really created or not after the command was executed. Likewise, not rely on the command to always return an error message or do the right thing. This means in reverse that I can omit the check of the interface command (I don't mean omitting the command call but 'expect' and '.resolves.not.toThrowError() around it'), because the result must create a folder or subfolder and that's what it's all about after the command has been executed.

However, I would leave both variants in the test, unless there is a best practice.

Please check my test again and give me feedback if I can improve something.